### PR TITLE
Add welcome tour to fraud protection settings

### PR DESCRIPTION
**Pull request by eduardoumpierre**
Fixes Automattic/woocommerce-payments-server#3083

#### Changes proposed in this Pull Request

Add the welcome tour component to the WooCommerce payments settings.

#### Testing instructions

* Checkout to this branch `add/3083-welcome-tour`
* Run `npm start`
* Enable the `wcpay_fraud_protection_settings_active` flag on your client
  * If you don't have it set yet, run: `npm run wp option add wcpay_fraud_protection_settings_active 1` 
  * Otherwise, update it by running: `npm run wp option update wcpay_fraud_protection_settings_active 1`
* Go to **Payments > Settings**
* The welcome tour should pop up
* Go through the tour steps and finish it – or click on the close button
* Refresh the page – it shouldn't pop up again
* Check the welcome tour flag by running `npm run wp option get wcpay_fraud_protection_welcome_tour_dismissed` – it should return `1`
* If you want to test it again, run the following command:  `npm run wp option update wcpay_fraud_protection_welcome_tour_dismissed 0`
* Refresh the page – the tour should pop up again

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/settings/fraud-protection/tour/index.tsx
Comment (jessepearson) [inline client/settings/fraud-protection/tour/index.tsx]: This is most likely a design thing, but I feel like these sentences do not read well.

Commas in English mean to pause while reading, or they can split up compound sentences. In this instance, I don't feel like a pause is needed, and it is not a compound sentence.

> Incoming transactions will now be screened for common risk factors, at the level of your choosing.

For the second sentence, "these filters," are mentioned, but filters are not mentioned in the first sentence.

> Review any transactions caught by these filters and select whether you'd like to approve or decline them

A suggestion to replace both sentences would be:

> You can now choose a level of protection to have for screening incoming transactions. You will then be able to review any caught transactions and select whether you would like to approve or decline them.

**Review by jessepearson — COMMENTED** [review]
Everything works as described, and I had one suggestion about some sentences.

React isn't my strength at the present moment, but everything looks straight forward and clean. We may need another set of eyes to take a look to make sure there wasn't anything I missed. 

After completing the steps, I got the below in my debug.log file. I don't think there's anything you can do about it due to it's likely a dependency issue, I just wanted to note it.

> [16-Mar-2023 12:20:32 UTC] The Automattic\WooCommerce\Admin\API\Options::update_options function is deprecated since version 3.1.
> [16-Mar-2023 12:20:32 UTC] The Automattic\WooCommerce\Admin\API\Options::update_options function is deprecated since version 3.1.
> [16-Mar-2023 12:20:32 UTC] The Automattic\WooCommerce\Admin\API\Options::update_options function is deprecated since version 3.1.
> [16-Mar-2023 12:20:32 UTC] The Automattic\WooCommerce\Admin\API\Options::update_options function is deprecated since version 3.1.

File: client/settings/fraud-protection/tour/index.tsx
Comment (eduardoumpierre) [reply client/settings/fraud-protection/tour/index.tsx]: Thanks for your suggestion @jessepearson!

@elizaan36 what do you think about this change?

Comment (eduardoumpierre) [reply client/settings/fraud-protection/tour/index.tsx]: As @elizaan36 is AFK, I'll proceed with the change. It was addressed on [6745aca](https://github.com/Automattic/woocommerce-payments/pull/5716/commits/6745aca1388460629b11952641aa93bcd301f7d4). Thanks again for your suggestion @jessepearson!

Comment (eduardoumpierre) [reply client/settings/fraud-protection/tour/index.tsx]: Regarding the deprecation logs, I didn't find a better way to do it, and there are some other occurrences of the same function to update the options.

I'll create an issue to document this behavior. Thanks for catching it!

**Comment by tpaksu** [comment]
Resolved the merge conflicts for you after I merged the settings screen to trunk. FYI.

File: client/settings/fraud-protection/tour/index.tsx
Comment (allie500) [inline client/settings/fraud-protection/tour/index.tsx]: It looks like the menu location is backwards here: `Transactions > Payments`. I believe it should be: `Payments > Transactions`.

wcpay-menu-payments-transactions

**Review by allie500 — CHANGES_REQUESTED** [review]
Thanks for working on this welcome tour, Eduardo! The code looks good and makes sense to me. Though, I'm still very new to React so as Jesse also mentioned, it may be good to get some more seasoned eyes on it before final approval. 

Just a note for the testing instructions. I don't know if it was just my local dev site or what, but to get the tour to work, I also needed to run `npm run build:client`. 

Lastly, the tour worked as described and makes sense. Though I did leave one inline comment related to the menu location in a string, it looks like it is backwards. Can you double check that?

File: client/settings/fraud-protection/tour/index.tsx
Comment (eduardoumpierre) [reply client/settings/fraud-protection/tour/index.tsx]: Nice catch! It was fixed on [d304eab](https://github.com/Automattic/woocommerce-payments/pull/5716/commits/d304eabfaa10334345eab99927a2fd68383e7dae).

**Comment by botwoo** [comment]
### Test the build on Jurassic Ninja 

This feature is only available for A12s. Please log in into https://jurassic.ninja/specialops/ before clicking the link below:

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/3083-welcome-tour) :rocket:

- Latest commit: e2420f73e3f46e6f1bd6c7f5caa1af7cac55d5d3
- Build time: 2023-03-22 11:32:31 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/fraud-protection/tour/index.tsx
Comment (eduardoumpierre) [reply client/settings/fraud-protection/tour/index.tsx]: Regarding the build command, it is required indeed. I'll update the PR description to have that information. Thanks!

Comment (allie500) [reply client/settings/fraud-protection/tour/index.tsx]: Thanks for fixing that string! LGTM

**Review by allie500 — APPROVED** [review]
Thanks for updating the string in the final tour card. Tests are passing for me. The code reads easily and makes sense. I think this is good to ship.

**Pull request merged**
